### PR TITLE
Adding option to set the cluster name and context on kubeconfig file

### PR DESCRIPTION
**Pull request by ktsakalozos**
Probably not the right target for this PR... regardless, have a look :)

### CODE REVIEW

File: cluster/juju/layers/kubernetes-master/config.yaml [hunk 1]
```diff
@@ -30,3 +30,7 @@ options:
     type: string
     default: "RBAC"
     description: kube-apiserver --authorization-mode <value>
+  cluster-context:
+    type: string
+    default: "juju-context"
```
Comment (marcoceppi) [inline cluster/juju/layers/kubernetes-master/config.yaml]: Could we leave this blank? And use the juju_model_name from env?

Comment (ktsakalozos) [inline cluster/juju/layers/kubernetes-master/config.yaml]: Agreed we should have the juju_model_name by default there. AFAIK there is no way to grab the model name from within a charm. What we could do is to have the conjure-up spell fill in the cluster-context config param. Would that be ok?

File: cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py [hunk 2]
```diff
@@ -709,17 +709,22 @@ def build_kubeconfig(server):
     cert_exists = cert and os.path.isfile(cert)
     # Do we have everything we need?
     if ca_exists and key_exists and cert_exists:
+        context = hookenv.config('cluster-context')
         # Cache last server string to know if we need to regenerate the config.
-        if not data_changed('kubeconfig.server', server):
+        # Same for cluster-context.
+        if (not data_changed('kubeconfig.server', server)
```
Comment (ktsakalozos) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: @chuckbutler should I remove the data_change guards?

Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/reactive/kubernetes_master.py]: I think its completely safe and reasonable to do so.

File: cluster/juju/layers/kubernetes-master/config.yaml (hunk 1 above)
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-master/config.yaml]: Actually @ktsakalozos you can get the model name from the hookenv.

https://jujucharms.com/docs/stable/reference-environment-variables#juju_model_name

Comment (ktsakalozos) [inline cluster/juju/layers/kubernetes-master/config.yaml]: Nice, didn't know about that!

**Comment by CalvinHartwell** [comment]
@tvansteenburgh it looks like @ktsakalozos already has a partial fix for [https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/330](https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/330) waiting to be merged. 

Can we merge this first and then I will check/verify/fix issue 330?

**Comment by marcoceppi** [comment]
If you do take up this merge request, we no longer base pull requests against this repo, make sure it's targeted and rebased off of github.com/kubernetes/kuberentes

**Comment by hyperbolic2346** [comment]
Closing this PR to clean up this repo so we can use it again.

**Pull request closed without merging**